recipe/meta.yaml

```diff
@@ -1,33 +1,80 @@
 {% set name = "scikit-optimize" %}
 {% set version = "0.9.0" %}
-{% set sha256 = "77d8c9e64947fc9f5cc05bbc6aed7b8a9907871ae26fe11997fd67be90f26008" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+    sha256: 77d8c9e64947fc9f5cc05bbc6aed7b8a9907871ae26fe11997fd67be90f26008
+    patches:
+      - 0001-replace-deprecated-np.int.patch
+      - 0002-use-compatible-mse-criteria-with-scikit-learn.patch
+  - url: https://github.com/{{ name }}/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+    sha256: b5602b22c7aacabe3afd36619f662d1f287f2e3fe9a5aa19bbc83f851b30b7bf
+    folder: git-repo
 
 build:
   number: 1
-  script: {{ PYTHON }} -m pip install . -vv
-  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
+  build:
+    - patch  # [not win]
+    - m2-patch  # [win] 
   host:
-    - python >=3.6
+    - python
     - pip
+    - wheel
+    - setuptools
   run:
-    - python >=3.6
-    - numpy
-    - scipy >=0.14.0
-    - scikit-learn >=0.19.1
-    - matplotlib-base
-    - pyaml
-    - joblib
+    - python
+    - {{ pin_compatible('numpy') }}
```

Comment:
according to global cbc the numpy is either 1.21 or 1.23 depending on python version, on upstream there is minimal version specified, but not maximal
https://github.com/scikit-optimize/scikit-optimize/blob/a2369ddbc332d16d8ff173b12404b03fea472492/requirements.txt#L1

what is the maximal version where np.int still works ? Cannot we use this version instead ?

Reply:
max version for np.int would be `numpy <1.20` which introduces a variety of other problems due to most dependencies being incompatible with such old `numpy`.